tests/integration/Product/ProductQueryRequestTest.php

```diff
@@ -304,4 +308,132 @@ function (Product $product) use ($client) {
+            function (Store $store) use ($client) {
+                ProductFixture::withDraftProduct(
+                    $client,
+                    function (ProductDraft $draft) {
+                        return $draft
+                            ->setName(LocalizedString::ofLangAndText(
+                                'de',
+                                'test-' . ProductFixture::uniqueProductString() . '-Name'
+                            )->add('en', 'test-' . ProductFixture::uniqueProductString() . '-name'));
+                    },
+                    function (Product $product) use ($client, $store) {
+                        $request = RequestBuilder::of()->productProjections()
+                            ->getById($product->getId())
+                            ->storeProjection($store->getKey())
+                            ->staged(true);
+                        $response = $this->execute($client, $request);
+                        $result = $request->mapFromResponse($response);
+
+                        $keyNameResult = key($result->getName()->toArray());
+
+                        $this->assertInstanceOf(ProductProjection::class, $result);
+                        $this->assertSame('en', $keyNameResult);
+                        $this->assertSame($product->getId(), $result->getId());
+                    }
+                );
+            }
+        );
+    }
+
+    public function testFallbackStoreProjectionGetById()
+    {
+        $client = $this->getApiClient();
+
+        StoreFixture::withDraftStore(
+            $client,
+            function (StoreDraft $storeDraft) {
+                return $storeDraft->setLanguages(['en']);
+            },
+            function (Store $store) use ($client) {
+                ProductFixture::withDraftProduct(
+                    $client,
+                    function (ProductDraft $draft) {
+                        return $draft
+                            ->setName(LocalizedString::ofLangAndText(
+                                'de',
+                                'test-' . ProductFixture::uniqueProductString() . '-Name'
+                            )->add('it', 'test-' . ProductFixture::uniqueProductString() . '-nome'));
+                    },
+                    function (Product $product) use ($client, $store) {
+                        $request = RequestBuilder::of()->productProjections()
+                            ->getById($product->getId())
+                            ->storeProjection($store->getKey())
+                            ->staged(true);
+                        $response = $this->execute($client, $request);
+                        $result = $request->mapFromResponse($response);
+
+                        $keyNameResult = key($result->getName()->toArray());
+
+                        $this->assertInstanceOf(ProductProjection::class, $result);
+                        $this->assertNotEquals(current($store->getLanguages()), $keyNameResult);
```

Comment:
What does the `current` call do?

Reply:
current is a php method which returns the current element in an array otherwise it will take the languages as an array and then I couldn't do the assertion